transmart-core-db/grails-app/services/org/transmartproject/db/userqueries/UserQuerySetService.groovy

```diff
@@ -209,36 +202,35 @@ class UserQuerySetService implements UserQuerySetResource {
         List<Long> removedIds = previousPatientIds - newPatientIds
 
         if (addedIds.size() > 0 || removedIds.size() > 0) {
-
-            QuerySet querySet = new QuerySet(
-                    query: query,
-                    setSize: newPatientIds.size(),
-                    setType: SetType.PATIENT
-            )
+            QuerySet querySet = new QuerySet()
+            querySet.query = query
+            querySet.setSize = (Long) newPatientIds.size()
+            querySet.setType = SetType.PATIENT
 
             List<QuerySetInstance> querySetInstances = []
             querySetInstances.addAll(newPatientIds.collect{
-                new QuerySetInstance(
-                        querySet: querySet,
-                        objectId: it,
-                )
+                QuerySetInstance setInstance = new QuerySetInstance()
+                setInstance.querySet = querySet
+                setInstance.objectId = it
+                setInstance
             })
 
             List<QuerySetDiff> querySetDiffs = []
             querySetDiffs.addAll(addedIds.collect {
-                new QuerySetDiff(
-                        querySet: querySet,
-                        objectId: it,
-                        changeFlag: ChangeFlag.ADDED
-                )
+                QuerySetDiff setDiff = new QuerySetDiff()
+                setDiff.querySet = querySet
+                setDiff.objectId = it
+                setDiff.changeFlag = ChangeFlag.ADDED
+                setDiff
             })
             querySetDiffs.addAll(removedIds.collect {
-                new QuerySetDiff(
-                        querySet: querySet,
-                        objectId: it,
-                        changeFlag: ChangeFlag.REMOVED
-                )
+                QuerySetDiff setDiff = new QuerySetDiff()
+                setDiff.querySet = querySet
+                setDiff.objectId = it
+                setDiff.changeFlag = ChangeFlag.REMOVED
+                setDiff
             })
+
             querySet.save(flush: true)
```

Comment:
Do we also want to have `failOnError: true` here, to better capture errors?

Reply:
hmm, but then maybe better to catch the errors and just log some warnings not to fail the scan for all queries, when there is some data issue for one query?